integration/build_test.go

```diff
@@ -108,6 +115,92 @@ func TestBuild(t *testing.T) {
 	}
 }
 
+//see integration/testdata/README.md for details
+func TestBuildInCluster(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	if !ShouldRunGCPOnlyTests() {
+		t.Skip("skipping test that is gcp only")
+	}
+
+	cleanupSkaffoldBinary := copySkaffoldBinary(t)
+	defer cleanupSkaffoldBinary()
+
+	suffix := uuid.New().String()
+	podName := fmt.Sprintf("skaffold-in-cluster-%s", suffix)
+	cleanupKustomization := setupSuffixKustomization(suffix, t)
+	defer cleanupKustomization()
+
+	const namespace = "default"
+
+	logs := skaffold.Run("-p", "create-build-step", "--cache-artifacts=true").InDir("./testdata/skaffold-in-cluster").InNs(namespace).RunOrFailOutput(t)
+	t.Logf("create-build-step logs: \n%s", logs)
+	defer func() {
+		if output, err := skaffold.Delete("-p", "create-build-step").InNs(namespace).InDir("./testdata/skaffold-in-cluster").RunWithCombinedOutput(t); err != nil {
+			t.Logf("failed to cleanup skaffold-in-cluster: %s, output: %s", err, output)
+		}
+	}()
+
+	client, err := kubernetes.Client()
```

Comment:
Other tests use `SetupNamespace` to create a ns and get a namespaced k8s client. Can this be done here?